src/pkg/services/m365/api/mail_test.go

```diff
@@ -370,34 +372,175 @@ func (suite *MailAPIIntgSuite) TestMail_attachmentListDownload() {
 	}
 }
 
-func (suite *MailAPIIntgSuite) TestMail_PostLargeAttachment() {
-	t := suite.T()
+type attachment struct {
+	name    string
+	data    string
+	isLarge bool
+}
 
-	ctx, flush := tester.NewContext(t)
-	defer flush()
+func createMailWithAttachment(
+	ctx context.Context,
+	t *testing.T,
+	ac Client,
+	userID string,
+	mailFolder graph.Container,
+	attachments ...attachment,
+) models.Messageable {
+	msg := models.NewMessage()
 
-	userID := tconfig.M365UserID(suite.T())
+	msg.SetSubject(ptr.To("attachment test"))
 
-	folderName := testdata.DefaultRestoreConfig("maillargeattachmenttest").Location
-	msgs := suite.its.ac.Mail()
-	mailfolder, err := msgs.CreateContainer(ctx, userID, MsgFolderRoot, folderName)
+	item, err := ac.Mail().PostItem(
+		ctx,
+		userID,
+		ptr.Val(mailFolder.GetId()),
+		msg)
 	require.NoError(t, err, clues.ToCore(err))
 
-	msg := models.NewMessage()
-	msg.SetSubject(ptr.To("Mail with attachment"))
+	for _, attach := range attachments {
+		if attach.isLarge {
+			id, err := ac.Mail().PostLargeAttachment(
+				ctx,
+				userID,
+				ptr.Val(mailFolder.GetId()),
+				ptr.Val(item.GetId()),
+				attach.name,
+				[]byte(attach.data))
+			require.NoError(t, err, clues.ToCore(err))
+			require.NotEmpty(t, id, "empty id for large attachment")
+		} else {
+			att := models.NewFileAttachment()
+
+			att.SetName(ptr.To(attach.name))
+			att.SetContentBytes([]byte(attach.data))
+
+			err = ac.Mail().PostSmallAttachment(
+				ctx,
+				userID,
+				ptr.Val(mailFolder.GetId()),
+				ptr.Val(item.GetId()),
+				att)
+			require.NoError(t, err, clues.ToCore(err))
+		}
+	}
 
-	item, err := msgs.PostItem(ctx, userID, ptr.Val(mailfolder.GetId()), msg)
-	require.NoError(t, err, clues.ToCore(err))
+	return item
+}
+
+func (suite *MailAPIIntgSuite) TestMail_PostAndGetAttachments() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
 
-	id, err := msgs.PostLargeAttachment(
+	userID := tconfig.M365UserID(t)
+	folderName := testdata.DefaultRestoreConfig("getattachmentstest").Location
+	mailFolder, err := suite.its.ac.Mail().CreateContainer(
 		ctx,
 		userID,
-		ptr.Val(mailfolder.GetId()),
-		ptr.Val(item.GetId()),
-		"raboganm",
-		[]byte("mangobar"))
+		MsgFolderRoot,
+		folderName)
 	require.NoError(t, err, clues.ToCore(err))
-	require.NotEmpty(t, id, "empty id for large attachment")
+
+	tests := []struct {
+		name          string
+		createMessage func(
+			ctx context.Context,
+			t *testing.T,
+			userID string) models.Messageable
+		verify func(t *testing.T, item models.Messageable)
+	}{
+		{
+			name: "Single attachment",
```

Comment:
nit: minor naming update so folks know it's a large attachment
```suggestion
			name: "Single large attachment",
```